### ISSUE TITLE
Tiles are loading but not displaying. 

### ISSUE THREAD

**Issue by christianroyal** [p-lr/MapCompose#53]
Any reason that tiles may be displaying like such.  They do show up as you zoom in but they disappear again. 
<img width="405" alt="Screen Shot 2022-02-25 at 5 36 44 PM" src="https://user-images.githubusercontent.com/52167242/155812581-4c60b453-b33e-409f-9041-d40c2eea1f5f.png">

**Comment by christianroyal** [comment]
W/BitmapFactory: bitmap marked for reuse (262144 bytes) can't fit new bitmap (524288 bytes) would this warning have anything to do with it?

**Comment by christianroyal** [comment]
<img width="382" alt="Screen Shot 2022-03-07 at 2 45 45 PM" src="https://user-images.githubusercontent.com/52167242/157106870-17277227-33af-4b7c-b785-639e910d50b0.png">
<img width="380" alt="Screen Shot 2022-03-07 at 2 46 09 PM" src="https://user-images.githubusercontent.com/52167242/157106881-960ec1da-fc18-4538-8e49-fb84be1e48d2.png">

Comparison of two different instances. The top is expected but I can't figure out what is causing the bottom to happen. I checked and the tiles are all there. I can reproduce it in the demo app as well still can't figure out what's going on to cause the issue. My thoughts on a solution is to just do the tiling on my end and not take them from my server.

**Comment by p-lr** [comment]
> would this warning have anything to do with it?

Sure it does. Are you using `highFidelityColors(false)` on your `MapState`?

**Comment by christianroyal** [comment]
Yes I am. I will share a fork of the demo project as I don't think I can share my company project due to work rules. I will add my snippit into a fork and raise a PR so you can look into it? 

 When I add the highFidelityColor(false) it changes to now doing this and showing more tiles overall but still missing some:

<img width="384" alt="Screen Shot 2022-03-07 at 5 23 15 PM" src="https://user-images.githubusercontent.com/52167242/157128084-4b426b32-262a-40cc-91f5-0f9e73f36f52.png">
<img width="385" alt="Screen Shot 2022-03-07 at 5 23 40 PM" src="https://user-images.githubusercontent.com/52167242/157128086-dbc712c9-dde6-4945-aa6d-71b94a820af1.png">



` maState = MapState(4, 5760, 3840, workerCount = 15, tileSize = 256) {
                highFidelityColors(false)
                maxScale(40f)
                scroll(0.5, 0.5)
                scale(0.0f)
            }.apply {
                request.filePath?.let { path -> addLayer(makeTileStreamProvider(path)) }
                shouldLoopScale = false
                enableRotation()
                disableMarkerDrag("marker$markerCount")
            }`

**Comment by p-lr** [comment]
Just create a branch on your fork, reproduce the issue and let me know which repo/branch to look at.

**Comment by christianroyal** [comment]
forked and created missingtilesbug branch, reproduced the issue please let me know if you need anything else from me.

Thank you!

**Comment by p-lr** [comment]
I could fix the problem. The main issue is your images have different color depth. Some are 64bit, others 32bit, and others 8bit..
This library is designed to reuse memory as much as possible. When it loads an image for the first time, it tries to reuse a memory slot used by a former image (which isn't needed anymore). That works as long as slots have the same size. Why isn't the case when images have different color depth.
To fix the problem, I changed the color depth to 32bit (which is more than enough to display a floor plan). I used ImageMagic, and ran the following command on each level directory:
```
convert *.png -set filename:base "%[basename]" png32:"%[filename:base]".png
``` 

Other issues:
* At level 0 your had a ".jpg" file.
* The configured size of the map was too big

If you set me as contributor to your fork, I can push the changes on a `fix-missing-tiles` branch.

**Comment by christianroyal** [comment]
Disregrard the .jpg file that was in error.

**Comment by p-lr** [comment]
I've pushed the branch `fix-missing-tiles`

**Comment by p-lr** [comment]
Or, if possible, have it changed server side.
